src/bionics.cpp

```diff
@@ -1532,9 +1521,22 @@ bool player::install_bionics( const itype &type, int skill_level )
                                               difficult );
     }
 
-    if( !query_yn(
-            _( "WARNING: %i percent chance of genetic damage, blood loss, or damage to existing bionics! Install anyway?" ),
-            100 - chance_of_success ) ) {
+    std::map<body_part, int> issues = bionic_installation_issues( bioid );
+    // show all requirements which are not satisfied
+    if( !issues.empty() ) {
+        std::string detailed_info;
+        for( auto& elem : issues ) {
+            //~ <Body part name>: <number of slots> more slot(s) needed.
+            detailed_info += string_format( _( "\n%s: %i more slot(s) needed." ),
+                                            body_part_name_as_heading( elem.first, 1 ).c_str(),
+                                            elem.second );
+        }
```

Comment:
You can probably reuse `list_occupied_bps()` (the code looks pretty similar) and write something like this:
```c++
popup( list_bionic_bps( issues, ... ) ... );
```